LeanKit-Labs/tag-release
# Feature pre-release promotion

Adding in the ability to promote a pre-release tag to an official release based on discussed guidelines of the promotion process.

Testing:

1. Link this PR's version of tag-release to your own local version.
2. Create two feature branches based off of master.
3. Make changes in both of these feature branches and create several pre-releases within each feature-branch using the pre-release command: `$ tag-release -p`.
4. Promote one of the feature-branches to an official release using `$ tag-release --promote` and follow the flow. There should be no conflicts.
5. Promote the second feature-branch to an official release `$ tag-release --promote` and follow the flow. THERE SHOULD BE CONFLICTS HERE!!!

Afterwards, follow this flow again but provide the tag at the command line: `$ tag-release --promote v1.1.1-feature.0`. You should also be able to provide it in this format as well: `$ tag-release --promote 1.1.1-feature.0`

Pre-Release CHANGELOG
1. Should no longer modify the CHANGELOG.md file when creating a pre-release.
2. Verify this by creating a new feature-branch, make a couple of commits and attempt to create a pre-release from this branch: `$ tag-release -p`
3. Should no longer see changes to the CHANGELOG.md file when finalizing the release.
